vendor/bat-native-ads/src/bat/ads/internal/base/url_util.cc

```diff
@@ -13,6 +13,11 @@
 
 namespace ads {
 
+bool SchemeIsSupported(const GURL& url) {
+  const std::string pattern = R"(brave://(wallet\z|wallet/.*|sync\z|sync/.*|rewards\z|rewards/.*|settings/searchEngines\z))";
+  return url.SchemeIs(url::kHttpsScheme) || RE2::FullMatch(url.spec(), pattern);
```

Comment:
Here's I'd suggest a more step-by-step approach:

1. If the scheme is HTTPS then return `true` immediately (i.e. optimized for the 99% case).
2. If the scheme is `brave`, then extract the hostname.
3. If the hostname is `wallet`, `sync` or `rewards` then return `true`.
4. If the hostname is `settings` then extract the path.
5. If the path is `searchEngines` then return `true`.
6. In all other cases, return `false`.

It's tempting to use a regexp as a shorthand here, but it's almost always a bad idea to use regular expressions to parse URLs because there are lots of ways to express URLs and get around these kinds of checks.

Reply:
Cleaned up, thanks